lib/Sema/TypeChecker.cpp

```diff
@@ -714,13 +714,14 @@ void swift::performTypeChecking(SourceFile &SF, TopLevelContext &TLC,
 
     typeCheckFunctionsAndExternalDecls(TC);
   }
-  MyTC.reset();
 
   // Checking that benefits from having the whole module available.
   if (!(Options & TypeCheckingFlags::DelayWholeModuleChecking)) {
     performWholeModuleTypeChecking(SF);
   }
 
+  MyTC.reset();
```

Comment:
Except we do perform checks that depend on a type checker here, namely walking the inheritance graph for decls defined in other files from the same module. As you can see there is a trivial test case and resulting crash that many users have been hitting.